src/Config.ts

```diff
@@ -0,0 +1,76 @@
+import { ClientConfig } from "pg";
+
+/**
+ * The configuration for schemalint.
+ */
+type Config = {
+  /**
+   * The connection configuration for the database.
+   * @see https://node-postgres.com/apis/client
+   */
+  connection: ClientConfig;
+  /**
+   * The plugins to be used.
+   */
+  plugins?: string[];
+  /**
+   * The rules to be used.
+   */
+  rules: Record<string, RuleConfig>;
+  /**
+   * The schemas to be linted.
+   */
+  schemas: SchemaConfig[];
+  /**
+   * The configuration for ignoring problems.
+   */
+  ignores?: IgnoreConfig[];
+};
+
+/**
+ * A schema to be linted.
+ */
+export type SchemaConfig = {
+  /**
+   * The name of the schema to be linted.
+   */
+  name: string;
+  /**
+   * The rules to be used spefically for this schema. These rules will be merged with the global rules.
+   */
+  rules?: Record<string, RuleConfig>;
+};
+
+/**
+ * A rule configuration. The first element is the severity of the rule, and the rest of the elements are the options for the rule.
+ */
+export type RuleConfig = [Severity, ...unknown[]];
+
+/**
+ * The severity of a rule. `off` means the rule is disabled, `error` means the rule is enabled.
+ */
+export type Severity = "off" | "error";
```

Comment:
In fact, any value other than `"error"` can be specified to disable the rule, but I chose the value `"off"` to match ESLint. Does it matter?